contracts/modules/ReserveAuction/V1/ReserveAuctionV1.sol

```diff
@@ -0,0 +1,527 @@
+    //     |    !                 |    | extend auction               |    !                                  |
+    //     |    !                 |<---'                              |    !                                  |
+    //     |    !                 |                                   |    !                                  |
+    //     |    !                 |----.                                   !                                  |
+    //     |    !                 |    | emit AuctionDurationExtended()    !                                  |
+    //     |    !                 |<---'                                   !                                  |
+    //     |    !~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!                                  |
+    //     |    !~[noop]~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!                                  |
+    //   Caller           ,-------+--------.                ,---------+----------.                  ,---------+---------.
+    //     ,-.            |ReserveAuctionV1|                |ERC721TransferHelper|                  |ERC20TransferHelper|
+    //     `-'            `----------------'                `--------------------'                  `-------------------'
+    //     /|\
+    //      |
+    //     / \
+    /// @notice Places a bid on the auction, holding the funds in escrow and refunding any previous bids
+    /// @param _tokenContract The address of the ERC-721 token
+    /// @param _tokenId The ID of the ERC-721 token
+    /// @param _amount The amount to bid
+    /// @param _finder The address of this bid's referrer
+    function createBid(
+        address _tokenContract,
+        uint256 _tokenId,
+        uint256 _amount,
+        address _finder
+    ) external payable nonReentrant {
+        Auction storage auction = auctionForNFT[_tokenContract][_tokenId];
+
+        require(auction.seller != address(0), "createBid auction doesn't exist");
+        require(block.timestamp >= auction.startTime, "createBid auction hasn't started");
+        require(_amount >= auction.reservePrice, "createBid must send at least reservePrice");
+        unchecked {
+            require(auction.firstBidTime == 0 || block.timestamp < (auction.firstBidTime + auction.duration), "createBid auction expired");
+            require(
+                _amount >= (auction.amount + ((auction.amount * MIN_BID_INCREMENT_PERCENTAGE) / 100)),
+                "createBid must send more than 10% of last bid amount"
+            );
+        }
+
+        // If first bid --
+        bool firstBid;
+        if (auction.firstBidTime == 0) {
+            // Store time of bid
+            auction.firstBidTime = block.timestamp;
+            // Mark as first bid
+            firstBid = true;
+            // Transfer NFT into escrow
+            erc721TransferHelper.transferFrom(_tokenContract, auction.seller, address(this), _tokenId);
```

Comment:
good question - this is by design. `safeTransferFrom` is protection on the user side to prevent sending NFTs to contracts that may not be able to handle them. in our case, since we know we're the recipient and can handle them, we can use the simpler `transferFrom` and save gas

Reply:
Ah, that context helps! Thank you @kulkarohan.

I was mulling over proposing a follow up change including an annotation there, but I think your input is enough for someone to look at this in the future and figure out why we use the simpler function. Considering this resolved 👍 

```suggestion
            // `transferFrom` is enough to move the NFT into escrow since the contract is the recipient
            erc721TransferHelper.transferFrom(_tokenContract, auction.seller, address(this), _tokenId);
```